atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/TableFactoryRenderer.java

```diff
@@ -15,28 +15,53 @@
  */
 package com.palantir.atlasdb.table.description.render;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.SortedMap;
+import java.util.stream.Collectors;
 
+import javax.annotation.Generated;
+import javax.lang.model.element.Modifier;
+
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Maps;
+import com.google.common.collect.Multimap;
 import com.palantir.atlasdb.keyvalue.api.Namespace;
 import com.palantir.atlasdb.table.description.TableDefinition;
+import com.palantir.atlasdb.table.generation.Triggers;
+import com.palantir.atlasdb.transaction.api.Transaction;
+import com.squareup.javapoet.AnnotationSpec;
+import com.squareup.javapoet.ArrayTypeName;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.JavaFile;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+import com.squareup.javapoet.WildcardTypeName;
 
 public class TableFactoryRenderer {
     private final String schemaName;
     private final String packageName;
     private final String defaultNamespace;
     private final SortedMap<String, TableDefinition> definitions;
+    private final ClassName tableFactoryType;
+    private final ClassName sharedTriggersType;
 
     public TableFactoryRenderer(String schemaName,
-                                String packageName,
-                                Namespace defaultNamespace,
-                                Map<String, TableDefinition> definitions) {
+            String packageName,
+            Namespace defaultNamespace,
+            Map<String, TableDefinition> definitions) {
         this.schemaName = schemaName;
         this.packageName = packageName;
         this.definitions = Maps.newTreeMap();
         this.defaultNamespace = defaultNamespace.getName();
+        this.tableFactoryType = ClassName.get(packageName, getClassName());
```

Comment:
Might be good to have a `public static Constructor` here.

Reply:
Added